scikeras/wrappers.py

```diff
@@ -645,21 +656,14 @@ def _fit(self, X, y, sample_weight=None, warm_start=False):
                     )
 
         # pre process X, y
-        X, extra_args = self.preprocess_X(X)
-        # update self.X_dtype_, self.X_shape_
-        for attr_name, attr_val in extra_args.items():
-            setattr(self, attr_name, attr_val)
-        y, extra_args = self.preprocess_y(y)
-        # update self.classes_, self.n_outputs_, self.n_classes_ and
-        #  self.target_type_
-        for attr_name, attr_val in extra_args.items():
-            setattr(self, attr_name, attr_val)
+        X = self.preprocess_X(X, reset=reset)
+        y = self.preprocess_y(y, reset=reset)
 
         # build model
-        if (not warm_start) or (not hasattr(self, "model_")):
+        if reset:
```

Comment:
I suggest using a method for `reset` with some renaming to be more clear; that way, no public APIs are cluttered.

``` python
class BaseWrapper(...):
    @property
    def initialized(self):
        return hasattr(self, "model_")
    
    def _fit(self, X, y=None):
        ...
        if not self.warm_start or not self.initialized:
            ...
```

Reply:
`reset` is not part of any public APIs, maybe I'm misunderstanding?